pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -119,19 +120,22 @@ func (c *controller) reconcilePermissionClaims(ctx context.Context, apiBinding *
 				break
 			}
 		}
+		ok, err := permissionclaims.ValidateClaim(acceptedPC.Group, acceptedPC.Resource, acceptedPC.IdentityHash, grsToBoundResource)
+		if !ok {
+			invalidClaims = append(invalidClaims, acceptedPC)
+			identityMismatch = true
+			klog.V(3).Infof("invalid claim %v - reason %v", acceptedPC, err)
+			continue
+		}
+
+		claim, gvr, err := c.createPermissionClaimHelper(acceptedPC)
+		if err != nil {
+			invalidClaims = append(invalidClaims, acceptedPC)
+			klog.Errorf("unable to create permission claim helper %v", err)
+			continue
+		}
+		syncClaims[gvr] = append(syncClaims[gvr], claim)
```

Comment:
does this mean we now always reconcile all claims? while in the past we were depending on `addedPermissionClaims` for that. But `addedPermissionClaims` is now cosmetics?

Reply:
Yeah, I don't think that we can do anything smarter until we get more info on the API to handle the informer is slow problem.

Reply:
Can you remind me about the race? And also doc it here in the code?

Reply:
I have update this section, and removed the sync claims